src/WebJobs.Script.Analyzers/WebJobsAttributeAnalyzer.cs

```diff
@@ -0,0 +1,141 @@
+
+    public static void VerifyWebJobsLoaded() 
+        {
+            var x = new JobHost(new OptionsWrapper<JobHostOptions>(new JobHostOptions()), null);
+        }
+
+        public override void Initialize(AnalysisContext context)
+        {
+            // https://stackoverflow.com/questions/62638455/analyzer-with-code-fix-project-template-is-broken
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+
+            VerifyWebJobsLoaded();
+
+            // Analyze method signatures.
+            context.RegisterSyntaxNodeAction(AnalyzeMethodDeclarationNode, SyntaxKind.MethodDeclaration);
+
+            // Hook compilation to get the assemblies' references and build the WebJob tooling interfaces.
+            context.RegisterCompilationStartAction(AnalyzeCompilation);
+        }
+
+        private void AnalyzeCompilation(CompilationStartAnalysisContext context)
+        {
+            var compilation = context.Compilation;
+
+            AssemblyCache.Instance.Build(compilation);
+            _tooling = AssemblyCache.Instance.Tooling;
+
+            // cast to PortableExecutableReference which has a file path
+            var x1 = compilation.References.OfType<PortableExecutableReference>().ToArray();
+            var webJobsPath = (from reference in x1
+                               where IsWebJobsSdk(reference)
+                               select reference.FilePath).SingleOrDefault();
+
+            if (webJobsPath == null)
+            {
+                return; // Not a WebJobs project.
+            }
+        }
+
+        private bool IsWebJobsSdk(PortableExecutableReference reference)
+        {
+            if (reference.FilePath.EndsWith("Microsoft.Azure.WebJobs.dll"))
+            {
+                return true;
+            }
+            return false;
+        }
+
+        // This is called extremely frequently
+        // Analyze the method signature to validate binding attributes + types on the parameters
+        private void AnalyzeMethodDeclarationNode(SyntaxNodeAnalysisContext context)
+        {
+            if (_tooling == null) // Not yet initialized
+            {
+                return;
+            }
+
+            var methodDecl = (MethodDeclarationSyntax)context.Node;
+            var methodName = methodDecl.Identifier.ValueText;
+
+            if (!HasFunctionNameAttribute(context, methodDecl))
+            {
+                return;
+            }
+
+            // TODO: Attribute validation
+        }
+
+        // First argument to the FunctionName ctor.
+        private string GetFunctionNameFromAttribute(SemanticModel semantics, AttributeSyntax attributeSyntax)
+        {
+            foreach (var arg in attributeSyntax.ArgumentList.Arguments)
+            {
+                var val = semantics.GetConstantValue(arg.Expression);
+                if (!val.HasValue)
+                {
+                return null;
```

Comment:
I think you can eliminate this return if you inverse the if to check for a value then return `return val.Value as string;` within the if condition, unless you're trying to not loop through everything and want to return early?

Reply:
I think we are trying to check only the first argument. IMHO, explicitly reading the first argument might be a bit more readable. Something like this.
````
if (attributeSyntax.ArgumentList.Arguments.Count == 0)
{
    return null; 
}

var firstArgument = attributeSyntax.ArgumentList.Arguments[0];
var val = semantics.GetConstantValue(firstArgument.Expression);

return val.Value as string;
````

Reply:
Shyju is correct; the loop will hit a return statement on its first iteration. I'm not sure why it was written the way it was, but his suggested implementation is more straightforward. I'll make the substitution.